### ISSUE TITLE
get_posts queries altered too frequently

### ISSUE THREAD

**Issue by stephywells** [wpsitecare/admin-content-labels#6]
The order_by_admin_label is causing any get_posts queries that include 'orderby' => 'title' to fail and return no results. Any post that doesn't have an _admin_content_label meta is excluded from the results.

$query->set( 'orderby', str_replace( 'title', 'meta_value title', $query->get( 'orderby' ) ) );
$query->set( 'meta_key', '_admin_content_label' );

This an be seen in action with Formidable installed. Create a form and go to the form settings -> form actions. Every time an action is created and saved, it vanishes from the page.

**Comment by robneu** [comment]
Thanks for the report!

I was aware of this issue, and the only workaround for it that I could find was to make sure all posts were given that meta value upon plugin installation and new post creation. I'm guessing that Formidable creates new posts in a non-standard way which causes them to avoid having the meta value applied?

Formidable probably should be excluded entirely just as ACF is, but I'm sure it's not the only plugin that causes issues. Are your post types public? One thing I considered doing in the past is ignoring all non-public post types, which would probably help avoid these situations at least a little bit.

If you have any other suggestions or ideas, please let me know. :)

**Comment by stephywells** [comment]
The post type is created in a standard way.
https://github.com/Strategy11/formidable-forms/blob/master/classes/controllers/FrmFormActionsController.php

Maybe any non-public or post types with no UI can be excluded?

**Comment by robneu** [comment]
Yeah, it looks like excluding those would solve the issue in this case. I'm guessing it would also solve issues with other plugins that I'm not yet aware of, so I'll go ahead and make that change.

The meta gets added on the `save_post` hook, so if you use another process to save your form data that would probably be the reason why the meta is never added. If you do save on that hook, then I'm not sure what the root of the problem is exactly.

**Comment by stephywells** [comment]
Yeah, the save_post hook should be triggered since saving is done using wp_insert_post. But if any posts are created before the Admin Content Labels plugin is active, the posts will immediately disappear with activation. This also sounds like a lot of overhead and database space to generate this post meta on every post whether it's used or not. Maybe the query can be changed to something more like this?
http://wordpress.stackexchange.com/questions/102447/sort-on-meta-value-but-include-posts-that-dont-have-one

**Comment by robneu** [comment]
It is quite a lot of overhead for something so trivial and it's definitely a total hack. Even the way the meta is added when the plugin is installed is pretty hacky. It was the end result of fighting with ordering for like three days.

I tried doing things like what's suggested in that stack exchange thread but nothing I tried worked correctly. No matter how I did it, if the meta wasn't there the ordering would be incorrect (sorted by title then meta or vice versa) or totally broken like it is with your forms right now.

It seems like it might be a problem with the way core handles ordering, but it's such an edge case it's probably not really worth doing anything about it... I'll take a look at the sorting options again with fresh  eyes and see if any alternatives to what I'm doing now will work.

In the meantime, if you see a better way to handle the ordering t hat will work without the crazy hacks I have in place now, please let me know. I'd love to fix this in a way that wasn't quite so crazy...

**Comment by robneu** [comment]
@stephywells That last commit should solve the issue at hand, but I'd really like to reduce the ridiculous overhead for this somehow. http://wordpress.stackexchange.com/a/141367 seems promising. I'll test an implementation of it soon. In the meantime, if you have a chance can you test this patch on your end?

I'm going to open a new issue for rewriting the query and getting rid of the meta value requirement.
